Modules/Scripted/DICOM/DICOM.py

```diff
@@ -8,6 +8,7 @@
 import slicer
 from slicer.util import settingsValue, toBool
 from slicer.ScriptedLoadableModule import *
+from slicer.translation import _
```

Comment:
I don't recall if we've discussed this before, but I wonder if underscore is the right symbol to use for this.  People often use it as a dummy variable and that would overwrite this import.  Using `tr` might be less error prone, or even `_tr` since that is unlikely to be used as a variable name.

Reply:
I don't like Python's choice of `_` as translation function name (it is hard to search for it, not self-explaining, [used for other purposes](https://stackoverflow.com/questions/5893163/what-is-the-purpose-of-the-single-underscore-variable-in-python/5893946#5893946)), but [it is the Python standard](https://docs.python.org/3/library/gettext.html#internationalizing-your-programs-and-modules) and it is widely used. If we want a Pythonic solution we cannot use Qt style or invent our own function name.

Specifically about the dummy variable issue: developers seem to agree that in programs that use internationalization, underscore should be used for translation and double-underscore can be used as dummy variable.

If we all agree with the above then I can add it to the Slicer coding guidelines.

Reply:
Since we aren't using `gettext` I'm not sure we need to be bound by its conventions.

If we think being pythonic is important, is there a way to make the underscore read-only so that trying to use it as a dummy variable raises an error?  I guess if they do change the value of underscore to some dummy value they will get an error when it is used as a function somewhere else, but that is indirect and not foolproof.

Reply:
An alternative would be to go all Qt. It would mean this syntax:

```
# in module main
translate = slicer.app.translate

...

def MyModuleWidget(...):

    def someFunction(self):
        ...
        myMessage = translate("MyModule", "Press any key to continue")
        myMessage = translate("MyModule", "Press {key} to continue").format(key=key)
        ...
```

The huge advantage would be that we could use pylupdate in pyside6 (can be pip-installed) for string extraction.

The disadvantages would be that pylupdate requires writing in the context name at every translation; and Qt standard is used much less frequently as Python standard.

Reply:
> requires writing in the context name at every translation

In a prior comment [^1], @pieper suggested to use a decorator. Instead I just resurrected the `slicer_lupdate_preprocess.py` script processing the AST and injecting the context based on the module name. See https://gist.github.com/jcfr/fc2bbb1e6b30d08c06f56d403f10145c

### Usage

```
!pip install astor

python slicer_lupdate_preprocess.py -i MyModule.py -o MyModule.lupdate.py
```

```diff
diff MyModule.py MyModule.lupdate.py
7,8c7,8
<         myMessage = _("Press any key to continue")
<         myMessage = _("Press {key} to continue").format(key=key)
---
>         myMessage = _('MyModule', 'Press any key to continue')
>         myMessage = _('MyModule', 'Press {key} to continue').format(key=key)
```

```
!pip install pyside6

$ pyside6-lupdate MyModule.lupdate.py -ts MyModule.ts -tr-function-alias translate+=tr,translate+=_
Updating 'MyModule.ts'...
    Found 2 source text(s) (2 new and 0 already existing)
```

```
$ cat MyModule.ts
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE TS>
<TS version="2.1">
<context>
    <name>MyModule</name>
    <message>
        <location filename="MyModule.tr.py" line="8"/>
        <source>Press any key to continue</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="MyModule.tr.py" line="8"/>
        <source>Press {key} to continue</source>
        <translation type="unfinished"></translation>
    </message>
</context>
</TS>
```

[^1]: https://github.com/Slicer/Slicer/pull/6050#pullrequestreview-819563767